src/main/java/com/powsybl/openloadflow/sensi/WoodburyEngine.java

```diff
@@ -105,22 +193,26 @@ public DenseMatrix run(DenseMatrix preContingencyStates) {
     }
 
     /**
-     * Calculate post-contingency states values using pre-contingency states values and some flow transfer factors (alphas).
-     * @return an array of post-contingency voltage angle states (which is the same modified pre-contingency java object).
+     * Calculate post-contingency and post-actions states values, using pre-contingency states values and some flow transfer factors (alphas).
+     * @return an array of post-contingency and post-actions voltage angle states (which is the same modified pre-contingency java object).
      */
     public double[] run(double[] preContingencyStates) {
```

Comment:
Is there a reason why you need two methods run ?
  - If there was only one method the problem of DenseMatrix allocation would be solved once. But here we see new cases where a densMatrix is created.
  -  It also creates a problem that two methods with the same name do different things. One will use tapPositionChangeElements if defined in the class, the other won't.
  - It also causes some low level code duplication (ex formulas of postContingencyValue using the Alpha.

Reply:
(in any case, to avoid conflicts - we should wait for a rebase on https://github.com/powsybl/powsybl-open-loadflow/pull/1111 - once merged - before working an a refacto)

Reply:
I'm closing that topic after the method renaming, although only one of the three points is adressed.
  - The duplicate code is small
  - the DenseMatrix allocation is only used in the optional fastDC SA and won't cause rregressions.